### PR TITLE
AUT-5768 - feature/AUT-5768-fix-quickstart-runner-container-name-conflict

### PR DESCRIPTION
**Pull request by RajmundNowicki**
## Description

Added mechanism to remove  container `quickstart-runner` to avoid:

> ERROR: for quickstart-runner  Cannot create container for service runner: Conflict. The container name "/quickstart-runner" is already in use

## Type of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Tests (extending the test suite)
- [X] Refactor (internal improvement that doesn't change product functionality)
- [ ] Other (if none of the other choices apply)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,12 +11,14 @@ run-%-local:
 	./scripts/wait.sh 
 	docker-compose -f docker-compose.$*.yaml up --no-build -d 
 	./scripts/wait.sh
+	docker ps -a
```
Comment (ikawalec) [inline Makefile:14]: the wait script already does `docker ps -a`

File: Makefile [hunk 2]
```diff
@@ -51,19 +53,22 @@ restart-acp:
 .PHONY: lint
 lint: start-runner
 	docker exec quickstart-runner sh -c "golangci-lint run --fix --deadline=5m ./..."
+	make stop-runner
```
Comment (ikawalec) [inline Makefile:56]: I wouldn't like to stop and start container on every `make lint`

File: Makefile [hunk 3]
```diff
@@ -51,19 +53,22 @@ restart-acp:
...
 
 .PHONY: clean
 clean: 
-	docker-compose -f docker-compose.acp.local.yaml -f docker-compose.build.yaml down --remove-orphans
+	docker-compose -f docker-compose.build.yaml down --remove-orphans
```
Comment (ikawalec) [inline Makefile:60]: is this also terminating ACP?

File: Makefile [hunk 4]
```diff
@@ -88,14 +93,21 @@ set-saas-configuration:
...
+	docker ps -a
+
+.PHONY: stop-runner
+stop-runner:
+	docker rm -f quickstart-runner
 
 .PHONY: generate-obuk-integration-spec
 generate-obuk-integration-spec: start-runner
 	./scripts/generate_bank_spec.sh uk
+	make stop-runner
```
Comment (ikawalec) [inline Makefile:105]: why this is necessary? I'd remove it here and leave only in necessary places

**Review by ikawalec — CHANGES_REQUESTED** [review]
Added some minor comments

File: Makefile (hunk 1 above)
Comment (RajmundNowicki) [reply Makefile:14]: Yes it does, but not printing in logs as a command output
image
image

I can add `echo docker ps -a` into wait script after the  `echo "All services are healthy"`

Comment (ikawalec) [reply Makefile:14]: It seems that this is for debugging only. If everything works, it should be not printed for the end-user.
Can we print it only for Jenkins?

Comment (RajmundNowicki) [reply Makefile:14]: OK
I'll move it into the Jenkinsfile

Comment (RajmundNowicki) [reply Makefile]: I've added a DEBUG environment variable to `Jenkinsfile` and I've updated `wait.sh` script and `make clean` operation.

File: Makefile (hunk 3 above)
Comment (RajmundNowicki) [reply Makefile:58]: This is a good question.
Locally `--remove-orphans` flag will do it, but on Jenkins I'm not sure - will check it.
E.g.: 
for  `quickstart-runner` container is not working remotely (locally is ok)

File: Makefile (hunk 2 above)
Comment (RajmundNowicki) [reply Makefile:54]: I did that to avoid container leftovers on pipeline - I can move it into Jenkinsfile.

File: Makefile (hunk 4 above)
Comment (RajmundNowicki) [reply Makefile:105]: ok

File: Makefile (hunk 3 above)
Comment (RajmundNowicki) [reply Makefile:58]: It is 
`Removing orphan container "quickstart-acp"` 
image

**Review by ikawalec — APPROVED** [review]

**Pull request merged**